lib/pages/surveydetail/survey_detail_page.dart

```diff
@@ -1,21 +1,62 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/di/di.dart';
 import 'package:flutter_survey/extensions/build_context_ext.dart';
+import 'package:flutter_survey/pages/surveydetail/survey_detail_state.dart';
+import 'package:flutter_survey/pages/surveydetail/survey_detail_view_model.dart';
 import 'package:flutter_survey/pages/surveydetail/widget/start_survey.dart';
 import 'package:flutter_survey/pages/uimodel/survey_ui_model.dart';
+import 'package:flutter_survey/usecase/get_survey_detail_use_case.dart';
+
+final surveyDetailViewModelProvider =
+    StateNotifierProvider.autoDispose<SurveyDetailViewModel, SurveyDetailState>(
+        (ref) {
+  return SurveyDetailViewModel(getIt.get<GetSurveyDetailUseCase>());
+});
+
+final _surveyStreamProvider = StreamProvider.autoDispose<SurveyUiModel>(
+    (ref) => ref.watch(surveyDetailViewModelProvider.notifier).surveyStream);
+
+class SurveyDetailPage extends ConsumerStatefulWidget {
+  @override
+  _SurveyDetailPageState createState() {
+    return new _SurveyDetailPageState();
+  }
+}
+
+class _SurveyDetailPageState extends ConsumerState<SurveyDetailPage> {
+  @override
+  void initState() {
+    super.initState();
+    Future.delayed(Duration.zero, () {
+      final survey = context.arguments as SurveyUiModel;
+      ref.read(surveyDetailViewModelProvider.notifier).loadSurveyDetail(survey);
+    });
+  }
 
-class SurveyDetailPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    final survey = context.arguments as SurveyUiModel;
-    return _buildSurveyPage(survey);
+    final uiModel = ref.watch(_surveyStreamProvider).value;
+    return ref.watch(surveyDetailViewModelProvider).when(
```

Comment:
A black screen shows a little time when we come to Survey Detail screen. I think it happens because we don't apply Material widget (Scaffold in this case) at the beginning time.
Should we add `Scaffold` as a parent widget and then return the body according to the result? 💭
 ```suggestion
    return Scaffold(
        body: ref.watch(surveyDetailViewModelProvider).when(...);
        );
```

Reply:
@manh-t you're right, I missed that wrapper 👌